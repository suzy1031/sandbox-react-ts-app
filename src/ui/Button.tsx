import { Button, Stack } from '@mui/material';
import { type ReactElement } from 'react';

const CustomButton = ({
  children,
  onClick,
  disabled,
}: {
  children?: React.ReactNode;
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
  disabled?: boolean;
}): ReactElement => {
  return (
    <Stack direction="row" spacing={2}>
      <Button color="secondary" variant="outlined" onClick={onClick}>
        left button
      </Button>
      <Button
        color="primary"
        variant="contained"
        disabled={disabled}
        onClick={onClick}
      >
        {children}
      </Button>
    </Stack>
  );
};
export default CustomButton;
