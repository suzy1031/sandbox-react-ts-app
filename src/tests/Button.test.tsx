import { render } from "@testing-library/react";
import Button from "../ui/Button";

test("renders correctly", () => {
  const { asFragment } = render(<Button>Test Button</Button>);
  expect(asFragment()).toMatchSnapshot();
});
