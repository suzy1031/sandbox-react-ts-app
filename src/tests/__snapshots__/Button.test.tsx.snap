// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-pl8gqh-MuiStack-root"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1l9sosn-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      secondary
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Test Button
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
